src/apis/core.ts

```diff
@@ -0,0 +1,53 @@
+import axios, { AxiosInstance, AxiosRequestConfig, Method } from 'axios';
+
+const HTTP_METHODS = {
+  GET: 'get',
+  POST: 'post',
+  PUT: 'put',
+  DELETE: 'delete',
+} as const;
+
+const TIME_OUT = 2000;
+
+const axiosInstance = axios.create({
+  baseURL: `${import.meta.env.VITE_API_URL}/${
+    import.meta.env.VITE_PORT_NUMBER
+  }`,
+  timeout: TIME_OUT,
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+axiosInstance.interceptors.request.use((config) => {
+  const accessToken = localStorage.getItem('auth-token');
+  if (accessToken) {
+    config.headers.Authorization = `Bearer ${accessToken}`;
+  }
+  return config;
+});
+
+axiosInstance.interceptors.response.use(
+  (response) => {
+    return response.data;
+  },
+  (error) => {
+    return Promise.reject(error);
+  },
+);
+
+const createApiMethod =
+  (axiosInstance: AxiosInstance, methodType: Method) =>
+  <T>(config: AxiosRequestConfig): Promise<T> => {
```

Comment:
```suggestion
  <T>(config: AxiosRequestConfig): Promise<AxiosResponse<T>> => {
```
실제로 이런 타입을 반환해주는 것 같은데 맞을까요?!?!

Reply:
이것저것 찾아봤는데 단순 Response타입으로 받는것과 AxiosResponse로 받는것의 차이점을 잘 모르겠네요.. 
<img width="526" alt="image" src="https://github.com/prgrms-fe-devcourse/FEDC5_NodakNodak_Noah/assets/102784200/94fb1ffd-26a8-437d-ab07-8e6d031074df">
해당 응답 타입에서 일단 사용될 내용은 data말곤 없다고 생각해서 T그대로 받아도 되지않을까 하는데 혹시 AxiosResponse타입을 썼을 때 다른 이점,,? 이 있을까요

Reply:
음.. 저도 찾아봤는데, 실제로 annotation 을 주지 않으면, 기본적으로  Promise<AxiosResponse<any,any>> 이런식으로 추론이 되는것같아요. (회색 부분이 추론된 타입입니다)

![image](https://github.com/prgrms-fe-devcourse/FEDC5_NodakNodak_Noah/assets/137467530/90ddeeb9-f56e-4f75-b0ad-fe9a3d944549)

![image](https://github.com/prgrms-fe-devcourse/FEDC5_NodakNodak_Noah/assets/137467530/957242db-f697-470b-86e6-b5343603b431)

그래서 객체로 무조건 한번 감싸져서 나오는거라, 만약 이렇게 하지 않으면, 사용시에, `api.get<AxsioResponse<ResponseType>>` 이런식으로 줘야할것 같아요!

Reply:
```ts
const createApiMethod =
  (axiosInstance: AxiosInstance, methodType: Method) =>
  <T>(config: AxiosRequestConfig): Promise<T> => {
    return axiosInstance({
      ...config,
      method: methodType,
    }).then(response => response.data);
  };
```
아핫 찾아봤는데 이런 방법도 있네요!! data 만 빼도록요!